### PR TITLE
Add extended capabilities for chat filters

### PR DESCRIPTION
**Pull request by bdach**
Port of https://github.com/ppy/osu-web/pull/11181. Shares the same test cases and end behaviour.

### CODE REVIEW

File: osu.Server.Spectator/ChatFilters.cs [hunk 1]
```diff
@@ -21,18 +28,57 @@ public ChatFilters(IDatabaseFactory factory)
 
         public async Task<string> FilterAsync(string input)
         {
-            if (filters == null)
+            if (!filtersInitialised)
+                await initialiseFilters();
+
+            if (blockRegex?.Match(input).Success == true)
+                throw new InvalidStateException(string.Empty);
```
Comment (bdach) [inline osu.Server.Spectator/ChatFilters.cs:35]: I didn't see much benefit to providing an actual exception message here.

This is how this looks client-side:

osu_2024-05-01_09-26-15

Comment (peppy) [reply osu.Server.Spectator/ChatFilters.cs:35]: How does it show if there is a message? Just as part of the exception description?

Comment (bdach) [reply osu.Server.Spectator/ChatFilters.cs:35]: Yeah it just gets appended at the end.

Comment (peppy) [reply osu.Server.Spectator/ChatFilters.cs:35]: Maybe better than nothing.. You just know users are going to report as a bug otherwise 😅

Comment (bdach) [reply osu.Server.Spectator/ChatFilters.cs:35]: Yeah sure. Not sure if you want me to apply something or are going to do so yourself, given the review self-request below.

Comment (peppy) [reply osu.Server.Spectator/ChatFilters.cs]: I'll grab it

**Review by peppy — APPROVED** [review]

**Pull request merged**